azurerm/internal/services/sentinel/sentinel_data_connector_microsoft_defender_advanced_threat_protection.go

```diff
@@ -0,0 +1,168 @@
+	if tenantId == "" {
+		tenantId = meta.(*clients.Client).Account.TenantId
+	}
+
+	param := securityinsight.MDATPDataConnector{
+		Name: &name,
+		MDATPDataConnectorProperties: &securityinsight.MDATPDataConnectorProperties{
+			TenantID: &tenantId,
+			DataTypes: &securityinsight.AlertsDataTypeOfDataConnector{
+				Alerts: &securityinsight.AlertsDataTypeOfDataConnectorAlerts{
+					State: securityinsight.Enabled,
+				},
+			},
+		},
+		Kind: securityinsight.KindMicrosoftDefenderAdvancedThreatProtection,
+	}
+
+	_, err = client.CreateOrUpdate(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name, param)
+	if err != nil {
+		return fmt.Errorf("creating Sentinel Data Connector Microsoft Defender Advanced Threat Protection %q: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceSentinelDataConnectorMicrosoftDefenderAdvancedThreatProtectionRead(d, meta)
+}
+
+func resourceSentinelDataConnectorMicrosoftDefenderAdvancedThreatProtectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.DataConnectorsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+	workspaceId := loganalyticsParse.NewLogAnalyticsWorkspaceID(id.SubscriptionId, id.ResourceGroup, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Sentinel Data Connector Microsoft Defender Advanced Threat Protection %q was not found - removing from state!", id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Sentinel Data Connector Microsoft Defender Advanced Threat Protection %q: %+v", id, err)
+	}
+
+	if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindMicrosoftDefenderAdvancedThreatProtection); err != nil {
+		return fmt.Errorf("asserting Sentinel Data Connector Microsoft Defender Advanced Threat Protection of %q: %+v", id, err)
+	}
+	dc := resp.Value.(securityinsight.MDATPDataConnector)
+
+	d.Set("name", id.Name)
+	d.Set("log_analytics_workspace_id", workspaceId.ID())
+	d.Set("tenant_id", dc.TenantID)
+
+	return nil
+}
+
+func resourceSentinelDataConnectorMicrosoftDefenderAdvancedThreatProtectionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sentinel.DataConnectorsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DataConnectorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Delete(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting Sentinel Data Connector Microsoft Defender Advanced Threat Protection %q: %+v", id, err)
```

Comment:
we can reduce this to:

```suggestion
		return fmt.Errorf("deleting %s: %+v", *id, err)
```